spec/class.dd

```diff
@@ -371,6 +371,9 @@ $(GNAME Constructor):
         }
         ------
 
+        $(P The compiler may generate a $(D __ctor) alias that matches with
```

Comment:
I'm not sure if the right wording is `may`, `should` or omit it. Should we enforce this to be implemented? I think so

Reply:
Is the compiler required to generate these aliases to comply to the spec? Or are they just an optional thing?

Reply:
> Is the compiler required to generate these aliases to comply to the spec? Or are they just an optional thing?

Technically yes, since e.g. `__ctor` is mentioned in several specification examples but this can be considered an implementation detail. You, as a user, probably don't require this, unless doing e.g. custom class allocators, custom traits or other advanced features. This is mostly used internally by the D runtime hooks and standard library traits, although I think it is useful to make it not implementation-defined to ease the user for the reasons I stated above.